src/anomalib/engine/engine.py

```diff
@@ -0,0 +1,189 @@
+"""Implements custom trainer for Anomalib."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from typing import Any
+
+from lightning import Callback
+from lightning.pytorch import LightningDataModule, Trainer
+from lightning.pytorch.callbacks import RichProgressBar
+from lightning.pytorch.trainer.connectors.callback_connector import _CallbackConnector
+from lightning.pytorch.utilities.types import _EVALUATE_OUTPUT, _PREDICT_OUTPUT, EVAL_DATALOADERS, TRAIN_DATALOADERS
+from omegaconf import DictConfig, ListConfig
+
+from anomalib.data import TaskType
+from anomalib.models import AnomalyModule
+from anomalib.post_processing import NormalizationMethod
+from anomalib.utils.callbacks import get_visualization_callbacks
+from anomalib.utils.callbacks.metrics import _MetricsCallback
+from anomalib.utils.callbacks.normalization import get_normalization_callback
+from anomalib.utils.callbacks.post_processor import _PostProcessorCallback
+from anomalib.utils.callbacks.thresholding import _ThresholdCallback
+from anomalib.utils.metrics.threshold import BaseThreshold, F1AdaptiveThreshold
+
+log = logging.getLogger(__name__)
```

Comment:
I don't think it does anymore since it is renamed to engine. But I am not sure how to detect conflicts. From what I see the logs from engine use Rich whereas the ones from trainer do not.

Reply:
super minor and nitpicky comment, but would it be an idea to use `logger` to match the consistency with the rest of the codebase? Alternatively, we could rename the other to `log` if you want to stick to `log`